### ISSUE TITLE
Catboost kill jupyter kernel if data contains NaN

### ISSUE THREAD

**Issue by mrk-andreev** [catboost/catboost#105]
Catboost kill jupyter kernel if data contains NaN. The program below kills the kernel:

```python
from catboost import CatBoostClassifier

x = [
    [float('nan'), float('nan')],
    [float('nan'), float('nan')]
]
y = [
    1,
    0
]

CatBoostClassifier().fit(x, y)
```

**Comment by vitekmel** [comment]
What is your version of python? And what OS?

**Comment by mrk-andreev** [comment]
Python 3.6.1 :: Anaconda custom (64-bit)
Ubuntu 16.04.3 LTS
Catboost version: 0.2 (latest)

**Comment by vitekmel** [comment]
Could you please try the same with "nan_mode='Min'"?

**Comment by vitekmel** [comment]
And did it throw any exceptions?

**Comment by mrk-andreev** [comment]
> Could you please try the same with "nan_mode='Min'"?
It work fine. No exceptions

> And did it throw any exceptions?
No, if I launch in ipython I got this:
```
libunwind: getEncodedP /home/donskov/arcadia/contrib/libs/libunwind_master/src/AddressSpace.hpp:269 - unknown pointer encoding
```

**Comment by vitekmel** [comment]
So, I don't understand. It is working with nan_mode='Min', but problem is that it doesn't throw anything with nan_mode='Forbidden' (that is default)? Do I understand properly?

**Comment by mrk-andreev** [comment]
Yes, you understood correctly.

**Comment by vitekmel** [comment]
Thanks, we have found the problem. Should be fixed in next version.

**Comment by vitekmel** [comment]
Should be fixed now, could you please recheck?

**Comment by mrk-andreev** [comment]
> Should be fixed now, could you please recheck?
Yes, it's works! Thanks!

**Comment by chengs** [comment]
in 0.9.1.1 it still exists when task_type=='GPU'

**Comment by annaveronika** [comment]
@chengs It probably was the case that you had nan in test dataset, but no nan in train dataset. There should be an exception in this case, but for some reason in gpu training sometimes there is kernel died instead of regular exception. Could you please check your data and say if it is the case?

**Comment by chengs** [comment]
@annaveronika, yes, you're right. 
but, sometimes it's ok even in the case you mentioned (usually in small datasets created by numpy.random).

**Comment by annaveronika** [comment]
Actually we get too many complains about this check, so we decided to remove it. In case of training where eval dataset has nans and train dataset has no nans there will be a warning.
Working on that right now, it'll be on github soon.

**Comment by annaveronika** [comment]
The fix is on github now! Please try version 0.10.2
